src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,784 @@
+            LocalDateTime startTime = LocalDateTime.parse(timeParts[0].trim(), dateTimeFormatter);
+            LocalDateTime endTime = LocalDateTime.parse(timeParts[1].trim(), dateTimeFormatter);
+            return new AddEventCommand(description, startTime, endTime);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date format. Please use yyyy-MM-dd HHmm format.");
+        }
+    }
+}
+
+/**
+ * Represents the list of tasks in the Duke application.
+ * Handles operations like adding, removing, and retrieving tasks from the list.
+ */
+class TaskList {
+    private ArrayList<Task> tasks;
+    /**
+     * Constructs a new TaskList instance with a pre-defined list of tasks.
+     *
+     * @param tasks The ArrayList of tasks to initialize the task list with.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+
+        this.tasks = tasks;
+    }
+    /**
+     * Constructs a new TaskList instance. Initializes the task list.
+     */
+    public TaskList() {
+
+        this(new ArrayList<>());
+    }
+    /**
+     * Adds a task to the task list.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+    /**
+     * Removes and returns a task from the task list at the specified index.
+     *
+     * @param index The index of the task to be removed.
+     * @return The removed task.
+     */
+    public Task removeTask(int index) {
+
+        return tasks.remove(index);
+    }
+    /**
+     * Retrieves a task from the task list at the specified index.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     */
+    public Task getTask(int index) {
+
+        return tasks.get(index);
+    }
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return The size of the task list.
+     */
+    public int getSize() {
+
+        return tasks.size();
+    }
+
+
+}
+
+
+enum TaskType {
+    TODO,
+    DEADLINE,
+    EVENT
+}
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
+
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    public String getStatusIcon() {
+
+        return "[" + taskType + "]" + (isDone ? "[X] " : "[ ] ");
+    }
+
+    public String getDescription() {
+
+        return description;
+    }
+
+    public void markAsDone() {
+
+        isDone = true;
+    }
+
+    public void markAsNotDone() {
+
+        isDone = false;
+    }
+    @Override
+    public String toString() {
+
+        return getStatusIcon() + getDescription();
+    }
+}
+
+class Todo extends Task {
+    public Todo(String description) {
+
+        super(description, TaskType.TODO);
+    }
+}
+
+class Deadline extends Task {
+    protected LocalDateTime by;
```

Comment:
Should the variable name be more descriptive as it is used throughout the class? Ex. deadline, target_date, etc.